classes/services/PKPFileService.inc.php

```diff
@@ -117,22 +117,35 @@ public function delete($id) {
 	 * This method sends a HTTP response and ends the request handling.
 	 * No code will run after this method is called.
 	 *
-	 * @param string $path The path to the file
+	 * @param string|int $pathOrFileId The path to the file or file ID
 	 * @param string $filename Filename to give to the downloaded file
 	 * @param boolean $inline Whether to stream the file to the browser
 	 */
-	public function download($path, $filename, $inline = false) {
+	public function download($pathOrFileId, $filename, $inline = false) {
+
+		$dispatcher = Application::get()->getRequest()->getDispatcher();
+
+		if (is_integer($pathOrFileId)) {
+			// Is the file ID
+			$fileId = $pathOrFileId;
+			$file = $this->get($fileId);
+			if (!$file) $dispatcher->handle404();
+			$path = $file->path;
+		} else {
+			// Is the path to the file; compatibility fix, see pkp/pkp-lib#6663
+			$path = $pathOrFileId;
+		}
 
 		if (!$this->fs->has($path)) {
-			Application::get()->getRequest()->getDispatcher()->handle404();
+			$dispatcher->handle404();
 		}
 
-		if (HookRegistry::call('File::download', [$path, &$filename, $inline])) {
+		if (HookRegistry::call('File::download', [$file ?? $path, &$filename, $inline])) {
```

Comment:
Don't modify the hook. We don't want the hooks to change between stable versions.

Reply:
Thanks! Fixed